src/util/database.connection.ts

```diff
@@ -0,0 +1,97 @@
+import { getConnectionManager, DataSource } from 'typeorm';
+import { SnakeNamingStrategy } from 'typeorm-naming-strategies';
+import { LoggerService } from '../logger/logger.service';
+import { ExecutionManager } from './execution.manager';
+
+/**
+ * Get connection based on the logged in tenant
+ * @returns connection
+ */
+
+export async function getConnection(): Promise<DataSource> {
```

Comment:
Can we try to use NestJS dependency injection in the request scope to obtain a connection from the connection pool? The connection can be injected to the necessary service as and when it needs it.

Reply:
Updated in commit [here](https://github.com/KeyValueSoftwareSystems/authentication-service/pull/80/commits/c985e3799d7895c07561e98edc98bbe04c241bf8)